### ISSUE TITLE
Offline search: allow POI fuzzy search 

### ISSUE THREAD

**Issue by bagage** [osmandapp/OsmAnd#5409]
- What version of OsmAnd are you using, on what device and Android/iOS version?
**OsmAnd+ 2.9.3 from 2018-02-08 on Android 7.0**

The auto-complete list of POI searchbar is too much sensitive, specially with accented characters (é È ï î). For instance in French, if you type `gite` instead of `gîte`, the latest won't show up. 

This is specially bad on mobile where you skip accented characters  on purpose due to poor accessibility. 

It would be great if either:

- OsmAnd could compare user input and presets by converting them to ASCII first (with possible losses which would not matter I believe) : `gîte` --> `gite`, `hôtel` -> `hotel`, etc.. 
- (better but harder) Instead of simply using `userInput.equals(presetName)`, a more evolved matching algorithm (similarly to iD, sublime text…) could rely on [Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance) or [fuzzy search](https://en.wikipedia.org/wiki/Approximate_string_matching), so that the system gets tolerant with mobile users. Eg `restorent` would still match `restaurant` and I think it would be OK.

edit: maybe should use [Collator class](https://developer.android.com/reference/java/text/Collator).

**Comment by vshcherb** [comment]
Ьукпув https://github.com/osmandapp/OsmAnd/issues/14198

**Comment by ryder-hook** [comment]
II don't understand why #14198 was closed. In my opinion, these two problems have nothing to do with each other. By the way, Bolanzo is the official Italian spelling for this town. There are no special characters in it (like hotel above).
The problem with #14198 is that Bolanzo is generally not found. No matter what street or place is searched for. And yes, my expectation is that the place will be found if I enter the official name of the town.

**Comment by scaidermern** [comment]
> II don't understand why #14198 was closed. In my opinion, these two problems have nothing to do with each other. By the way, Bolanzo is the official Italian spelling for this town.

No, the official spelling is Bolzano, not Bolanzo :) And this is one the reasons why this issue is related to your other issue. Fuzzy searches allow for spelling mistakes.

**Comment by ryder-hook** [comment]
You are correct in that I gave the wrong spelling before. But that doesn't change my assessment, see the following screenshot. I searched by address and narrowed it down to cities. I am currently about 270 km away from Bolzano. In this overview, Bolzano should now appear between X and Y i
![IMG_3600](https://user-images.githubusercontent.com/104902502/167563300-c310b0a1-7361-42d8-a27b-729f65401bd8.PNG)
n the search, but it does not.

**Comment by scaidermern** [comment]
Is this iOS? Looks a little bit strange to me.

For me on Android, searching for "Bol" (or Bolzano) correctly shows "Bolzano - Bozen" as a result.

**Comment by ryder-hook** [comment]
Yes it is iOS, newest version and newest iPhone.

**Comment by barathrm** [comment]
I'd be interested to know why this is considered "nice to have" only? In my experience everyone now expects fuzzy search from being trained by google to not having to remember names *exactly*. Or another example would be when I couldn't find an address in Osmand today because the website I'd gotten the address from had the address written in a slightly different way than the way it's published on Osmand. It couldn't find the address, even though only one letter was missing. With fuzzzy search, this would've worked. I must assume that a lot of people's search frustrations with Osmand could be solved with this.

**Comment by antoine-gu** [comment]
Absolutely necessery. I have searched for greisslerei and didnt get result for Stadtgreisslerei. Well how could i know that someone named it that way.
How could that be implemented. I think it would be great to have the possibility to choose between different searching algorithms or even add your own as you can add your own map sources.

**Comment by rriemann** [comment]
When nI have issues finding places in Osmand, I turn to Google Maps. Fuzzy Search can help here. I find it very important.

**Comment by scaidermern** [comment]
Maybe for fuzzy search the online search engine Photon could be used.

**Comment by Drup** [comment]
Small bump: this is basically the n°1 reason that makes it hard to recommend OsmAnd for others: you have to keep using google map to find actual shops and locations (since you overhead the name, and it's regularly "close but not quite correct").

This is also much worse in non-english languages.

**Comment by paalwilliams** [comment]
definitely agree. i find the search pretty difficult to use, even though the POI data is pretty complete, i have a difficult time finding stuff.
